File: 0009-palindrome-number/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x>=0 && x <= 9) return true
    if(x<0) return false
    if(x%10 == 0) return false
    
    let temp = 0, curr = x;
    while(x){
        temp = temp * 10 + x % 10;
        x = Math.floor(x/10);
    }
    return curr == temp;
};

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        if x < 10:
            return True
        
        
        temp, y = x, 0
        while x > 0:
            y = y * 10 + x % 10
            x = x // 10
            
        return y == temp
        